jackson/src/main/java/com/lightbend/lagom/javadsl/jackson/JacksonSerializerFactory.java

```diff
@@ -138,55 +140,99 @@ public MessageEntity deserialize(ByteString bytes) {
+            return deserializer;
+        }
 
-    @Override
-    public NegotiatedSerializer<MessageEntity, ByteString> serializerForResponse(
-        List<MessageProtocol> acceptedMessageProtocols) {
-      return serializer;
-    }
+        @Override
+        public NegotiatedSerializer<MessageEntity, ByteString> serializerForResponse(
+            List<MessageProtocol> acceptedMessageProtocols) {
+            return serializer;
+        }
 
-    private class DoneSerializer implements NegotiatedSerializer<MessageEntity, ByteString> {
+        private class DoneSerializer implements NegotiatedSerializer<MessageEntity, ByteString> {
 
-      private final ByteString doneJson = ByteString.fromString("{ \"done\" : true }");
+            private final ByteString doneJson = ByteString.fromString("{ \"done\" : true }");
 
-      @Override
-      public MessageProtocol protocol() {
-        return new MessageProtocol(Optional.of("application/json"), Optional.of("utf-8"), Optional.empty());
-      }
+            @Override
+            public MessageProtocol protocol() {
+                return new MessageProtocol(Optional.of("application/json"), Optional.of("utf-8"), Optional.empty());
+            }
+
+            @Override
+            public ByteString serialize(MessageEntity obj) {
+                return doneJson;
+            }
+        }
 
-      @Override
-      public ByteString serialize(MessageEntity obj) {
-        return doneJson;
-      }
+        private class DoneDeserializer implements NegotiatedDeserializer<MessageEntity, ByteString> {
+            @SuppressWarnings("unchecked")
+            @Override
+            public MessageEntity deserialize(ByteString bytes) {
+                return (MessageEntity) Done.getInstance();
+            }
+        }
     }
 
-    private class DoneDeserializer implements NegotiatedDeserializer<MessageEntity, ByteString> {
-      @SuppressWarnings("unchecked")
-      @Override
-      public MessageEntity deserialize(ByteString bytes) {
-        return (MessageEntity) Done.getInstance();
-      }
+    private class NOPMessageSerializer<MessageEntity> implements StrictMessageSerializer<MessageEntity> {
+
+        private final NegotiatedSerializer<MessageEntity, ByteString> serializer = new NOPSerializer();
+        private final NegotiatedDeserializer<MessageEntity, ByteString> deserializer = new NOPDeserializer();
+
+        private class NOPSerializer implements NegotiatedSerializer<MessageEntity, ByteString> {
+
+            @Override
+            public MessageProtocol protocol() {
+                return defaultProtocol;
+            }
+
+            @Override
+            public ByteString serialize(MessageEntity obj) {
+                return (ByteString) obj;
+            }
+        }
+
+        private class NOPDeserializer implements NegotiatedDeserializer<MessageEntity, ByteString> {
+            @Override
+            @SuppressWarnings("unchecked")
```

Comment:
If `(MessageEntity) bytes` is an unchecked cast, it's not safe, right?

Reply:
Practically not. This case equivalent to current `DoneDeserializer`. The compiler don't know, but we know that `DoneDeserializer` used only if `MessageEntity` is `Done.class` and `NOPDeserializer` used only if `MessageEntity` is `ByteString.class`.

Reply:
I played with the code so I see what you mean now 😄.  I moved the cast closer to where the assertion is made, so it's safer now.

Reply:
👍